src/features/defi/providers/univ2/hooks/useUniV2LiquidityPool.ts

```diff
@@ -146,8 +248,8 @@ export const useUniV2LiquidityPool = ({
             const contractAddress = fromAssetId(uniswapV2Router02AssetId).assetReference
             return await adapter.buildCustomTx({
               to: contractAddress,
-              // the eth value need to be starting with 0x and be base 16
-              value: '0x' + bnOrZero(value).toString(16),
+              // the ETH value need to be starting with 0x and be base 16
+              value: value === '0' ? '0' : '0x' + bnOrZero(value).toString(16),
```

Comment:
how does this do base 16?

Reply:
Yes, I can explain how this code converts a string to base 16.

First, let's define what base 16 (also known as hexadecimal) is. In base 16, there are 16 digits: 0, 1, 2, 3, 4, 5, 6, 7, 8, 9, A, B, C, D, E, and F. The value of each digit is determined by its position in the number, starting from the rightmost digit. The rightmost digit has a value of 16^0, the next digit to the left has a value of 16^1, the next has a value of 16^2, and so on.

Now, let's take a look at the code you provided:

`value: value === '0' ? '0' : '0x' + bnOrZero(value).toString(16),`

The `bnOrZero(value)` part suggests that `value` is a BigNumber object, which is a library for handling arbitrary-precision numbers in JavaScript. The `toString(16)` method call on this object returns a string representation of the number in base 16. 

So, this code takes the input `value`, converts it to a BigNumber object (if it isn't one already), and then converts that object to a string in base 16 using the `toString(16)` method. The resulting string is then concatenated with `'0x'` (which is the prefix used to indicate that the value is in base 16 in many programming languages, including JavaScript), to give a string in the format `'0x<hex string>'`.

For example, if `value` is equal to 255, then the resulting string would be `'0xFF'`. The `'0x'` prefix indicates that the value is in base 16, and the `'FF'` represents the hexadecimal value of 255.

Reply:
tl;dr https://mikemcl.github.io/bignumber.js/#toS

Reply:
derp it takes a base, ty